app/controllers/labels_controller.rb

```diff
@@ -3,7 +3,21 @@
 class LabelsController < ApplicationController
   load_and_authorize_resource
 
+  before_action :sanitize_page_param, only: :search
   def search
-    @labels = Label.order(:name)
+    results = Label.ransack(name_i_cont: params[:search]).result
+    paginated = results.paginate(per_page: 3, page: @page)
+
+    render json: {
+      results: paginated.map {|l| {id: l.name, text: l.name, label_color: l.color, label_font_color: l.font_color} },
+      pagination: {more: (@page * 3 < results.length)},
+    }
+  end
+
+  def sanitize_page_param
```

Comment:
Is there a specific reason you introduced this method? Sure, it will catch the case where no page number or a negative one is given, but is this likely to occur? There is one case you aren't checking so far: Someone could also get a page number that is larger than the total number of available pages. For the latter, WillPaginate already comes with a [helper method](https://github.com/mislav/will_paginate/blob/8113fddf4b506867e9af3aa59fe2df314b79860c/lib/will_paginate/collection.rb#L29-L35).

Since we no longer need the `@page` variable twice, I would suggest to just use the return value of this method (and without it being a before filter) if we want to keep it. Then, we should probably also add a test case 😉.

Reply:
We could probably just remove the function and have rails give out an error if no page parameter is given. I did not feel comfortable have a missing parameter lead to an error but that is probably okay. 
Maybe it is even better not having the function because an error is better than silently correcting something without the caller knowing.

Reply:
I just went ahead and removed the function but I am also happy to re-add it if you want.

Reply:
I think, it's fine the way it is. If no page parameter is given, WillPaginate will automatically assume page 1, similar to the method you previously added.